server/src/repositories/oauth.repository.ts

```diff
@@ -40,8 +40,8 @@ export class OAuthRepository {
       redirect_uri: redirectUrl,
       scope: config.scope,
       state,
-      code_challenge: codeChallenge,
-      code_challenge_method: 'S256',
+      code_challenge: client.serverMetadata().supportsPKCE() ? codeChallenge : '',
+      code_challenge_method: client.serverMetadata().supportsPKCE() ? 'S256' : '',
```

Comment:
Is this correct? I would have guessed undefined would be used to omit the property

Reply:
Me too but it's typed as URL query params.